src/screencast.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::HashMap;
+
+use zbus::dbus_interface;
+
+use zbus::zvariant::{DeserializeDict, ObjectPath, OwnedValue, SerializeDict, Type, Value};
+
+use enumflags2::BitFlags;
+
+use serde::{Deserialize, Serialize};
+
+use crate::request::RequestInterface;
+use crate::session::{append_session, CursorMode, PersistMode, Session, SourceType, SESSIONS};
+
+#[derive(SerializeDict, DeserializeDict, Type, Debug, Default)]
+/// Specified options for a [`Screencast::create_session`] request.
+#[zvariant(signature = "dict")]
+struct SessionCreateResult {
+    handle_token: String,
+}
+
+#[derive(SerializeDict, DeserializeDict, Type, Debug, Default)]
+/// Specified options for a [`Screencast::select_sources`] request.
+#[zvariant(signature = "dict")]
+pub struct SelectSourcesOptions {
+    /// A string that will be used as the last element of the handle.
+    /// What types of content to record.
+    pub types: Option<BitFlags<SourceType>>,
+    /// Whether to allow selecting multiple sources.
+    pub multiple: Option<bool>,
+    /// Determines how the cursor will be drawn in the screen cast stream.
+    pub cursor_mode: Option<CursorMode>,
+    pub restore_token: Option<String>,
```

Comment:
the key is a little different from xdg-desktop-portal document, but the code is from ashpd.. so maybe they are wrong? but it is optional, and we do not need it now, so it is safe